sonar-text-plugin/src/main/java/org/sonar/plugins/common/TextAndSecretsSensor.java

```diff
@@ -175,9 +179,10 @@ protected List<Check> getActiveChecks() {
 
   protected void initializeSpecificationBasedChecks(List<Check> checks) {
     SpecificationLoader specificationLoader = new SpecificationLoader();
+    Map<InputFileContext, List<TextRange>> reportedIssuesForCtx = new HashMap<>();
```

Comment:
I can see that we perform active checks sequentially, but can it be possible that we change it to parallel and would need a `ConcurrentHashMap` here?